lib/dma.h

```diff
@@ -174,7 +174,12 @@ void
 dma_unmap_region(dma_memory_region_t *region, void *virt_addr, size_t len);
 
 static inline int
-dma_map_sg(dma_controller_t *dma, int prot,
+dma_map_sg(dma_controller_t *dma,
+#if DMA_MAP_FAST_IMPL
```

Comment:
Should we get rid of DMA_MAP_FAST_IMPL? I do not understand what is the use of it.
With DMA_MAP_FAST_IMPL disabled I do not know how the dma_mapping will work.

Reply:
`DMA_MAP_FAST_IMPL` is the default behavior, see the top of the file for more details. In any case this is beyond the scope of this patch.

Reply:
Yes I had a look into the comment, the comment and the macros looks misleading it also needs to be fixed. Also It does not talks about DMA_MAP_FAST_IMPL.
Anyways I think currently with DMA_MAP_FAST_IMPL disabled, DMA_MAPS will not work, so these ifdefs with macros are not required and we can just get rid of it.
If you agree I will fix this and send the patch around.

Reply:
sure